test/common/init/target_impl_test.cc

```diff
@@ -60,6 +62,110 @@ TEST(InitTargetImplTest, ReadyWhenWatcherUnavailable) {
   EXPECT_FALSE(target.ready());
 }
 
+TEST(SharedTargetImplTest, OnSyncManagers) {
+  ManagerImpl m1("m1");
+  ManagerImpl m2("m2");
```

Comment:
Don't use `ManagerImpl`s in these tests, just use `ExpectableWatcherImpl`s as in the tests above. (Don't forget to remove the build dependency on init:manager_lib).

Reply:
Is it `program to an interface` thing? Not sure because I need a `Manager` here. Will look at code and ask

Reply:
No, you don't need any kind of `Manager` here at all. You just need to call `initialize` with a `Watcher` (preferably an `ExpectableWatcherImpl`) the same way that `Manager` does.